core/src/main/java/org/phoenixframework/liveview/lib/Node.kt

```diff
@@ -3,36 +3,47 @@ package org.phoenixframework.liveview.lib
 /** This class represents the valid node types of a `Document` tree */
 sealed class Node {
 
-    /** A marker node that indicates the root of a document
-     * A document may only have a single root, and it has no attributes
+    /**
+     * A marker node that indicates the root of a document A document may only have a single root,
+     * and it has no attributes
      */
     object Root : Node()
 
-    /**A typed node that can carry attributes and may contain other nodes*/
-    data class Element(val namespace: String, val tag: String, val attributes: Array<Attribute>) :
-        Node() {
-        internal var nativeObject: Long = 0
+    /** A typed node that can carry attributes and may contain other nodes */
+    class Element: Node {
+        private var nativeObject: Long
 
-        private external fun drop(pointer: Long)
+        internal constructor(pointer: Long) {
+            nativeObject = pointer
+        }
+
+        fun namespace(): String {
+            return get_namespace(nativeObject)
+        }
+
+        fun tag(): String {
+            return get_tag(nativeObject)
+        }
 
-        @Synchronized
-        private fun delete() {
-            if (nativeObject != 0L) {
-                drop(nativeObject)
-                nativeObject = 0
-            }
+        fun attributes(): Array<Attribute> {
+            return get_attributes(nativeObject)
         }
```

Comment:
This can remain a property with a getter defined

```suggestion
        val attributes(): Array<Attribute>
            get() = get_attributes(nativeObject)
```